pkg/admiralctl/migrate/migrate.go

```diff
@@ -0,0 +1,694 @@
+		policyName = names.GeneratePolicyName(o.Namespace, o.name, o.gvk.String())
+	} else {
+		policyName = o.PolicyName
+	}
+
+	_, err := kubeadmiralClient.CoreV1alpha1().PropagationPolicies(o.Namespace).Get(context.TODO(), policyName, metav1.GetOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
+		pp := buildPropagationPolicy(policyName, o.Namespace, o.Cluster, o.SchedulingMode)
+		_, err = kubeadmiralClient.CoreV1alpha1().PropagationPolicies(o.Namespace).Create(context.TODO(), pp, metav1.CreateOptions{})
+
+		return policyName, err
+	}
+	if err != nil {
+		return "", fmt.Errorf("failed to get PropagationPolicy(%s/%s) in control plane: %w", o.Namespace, policyName, err)
+	}
+
+	// PropagationPolicy already exists, not to create it
+	return "", fmt.Errorf("the PropagationPolicy(%s/%s) already exist, please edit it to propagate resource", o.Namespace, policyName)
+}
+
+// createClusterPropagationPolicy create ClusterPropagationPolicy in Kubeadmiral control plane
+func (o *CommandMigrateOption) createClusterPropagationPolicy(kubeadmiralClient *fedclient.Clientset) (string, error) {
+	var policyName string
+	if o.PolicyName == "" {
+		policyName = names.GeneratePolicyName("", o.name, o.gvk.String())
+	} else {
+		policyName = o.PolicyName
+	}
+
+	_, err := kubeadmiralClient.CoreV1alpha1().ClusterPropagationPolicies().Get(context.TODO(), policyName, metav1.GetOptions{})
+	if err != nil && apierrors.IsNotFound(err) {
+		cpp := buildClusterPropagationPolicy(policyName, o.Cluster, o.SchedulingMode)
+		_, err = kubeadmiralClient.CoreV1alpha1().ClusterPropagationPolicies().Create(context.TODO(), cpp, metav1.CreateOptions{})
+
+		return policyName, err
+	}
+	if err != nil {
+		return "", fmt.Errorf("failed to get ClusterPropagationPolicy(%s) in control plane: %w", policyName, err)
+	}
+
+	// ClusterPropagationPolicy already exists, not to create it
+	return "", fmt.Errorf("the ClusterPropagationPolicy(%s) already exist, please edit it to propagate resource", policyName)
+}
+
+// preprocessResource delete redundant fields to convert resource as template
+func preprocessResource(obj *unstructured.Unstructured) error {
+	// remove fields that generated by kube-apiserver and no need(or can't) propagate to member clusters.
+	if err := removeIrrelevantField(obj); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// removeIrrelevantField used to remove fields that generated by kube-apiserver and no need(or can't) propagate to
+// member clusters.
+func removeIrrelevantField(workload *unstructured.Unstructured, extraHooks ...func(*unstructured.Unstructured)) error {
+	// populated by the kubernetes.
+	unstructured.RemoveNestedField(workload.Object, "metadata", "creationTimestamp")
+
+	// populated by the kubernetes.
+	// The kubernetes will set this fields in case of graceful deletion. This field is read-only and can't propagate to
+	// member clusters.
+	unstructured.RemoveNestedField(workload.Object, "metadata", "deletionTimestamp")
```

Comment:
It is strange to federalize the resources in deleting status. Should we forbid the operation in preflight？/cc @gary-lgy @limhawjia